src/main/java/org/spongepowered/api/keyboard/KeyBinding.java

```diff
@@ -96,6 +98,34 @@ static Builder builder() {
         Builder displayName(Text displayName);
 
         /**
+         * Sets the press executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding pressed.
+         *
+         * @param executor The press executor
+         * @return This builder for chaining
+         */
+        Builder pressExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
+
+        /**
+         * Sets the release executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding release.
+         *
+         * @param executor The release executor
+         * @return This builder for chaining
+         */
+        Builder releaseExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
+
+        /**
+         * Sets the tick executor of this key binding, it will be called
+         * when a specific {@link Player} the key binding holds the key
+         * and a tick occurred.
+         *
+         * @param executor The tick executor
+         * @return This builder for chaining
+         */
+        Builder tickExecutor(@Nullable BiConsumer<Player, KeyBinding> executor);
```

Comment:
@kashike I think that in this case you are most likely going to use the `reset` anyway because the only setting that you want to keep is the `category`, the other ones are most likely to change, like the id and display name which should be unique and I assume that you won't reuse the same executors for multiple key bindings.